### PR TITLE
Add NixOS Configuration CI Tests

### PR DESCRIPTION
**Pull request by grantbevis**
The added GitHub Action will test the NixOS configuration file successfully builds. This useful for if there's a typo or another syntax problem with the file.

I have also included the dependabot GitHub action in this PR - I will delete the other PR that was just for Dependabot

### CODE REVIEW

**Comment by b3vis** [comment]
@Trundle / @Mix92 Could I bug you guys to review this PR? I'm keen to help build upon Trundle's fantastic work

**Comment by Mic92** [comment]
The eval error looks legit.

File: .github/workflows/update-flakes.yml [hunk 1]
```diff
@@ -11,20 +11,12 @@ jobs:
...
-        with:
-          install_url: https://github.com/numtide/nix-flakes-installer/releases/download/nix-2.4pre20201221_9fab14a/install
-          # Configure Nix to enable flakes
-          extra_nix_config: |
-            experimental-features = nix-command flakes
+        uses: cachix/install-nix-action@v16
 
       - name: Update flakes
-        run: nix flake update --recreate-lock-file
+        run: nix flake update
```
[the remark this answers is not in the document]
Comment (b3vis) [reply .github/workflows/update-flakes.yml:19]: Not a bad shout I hadn’t see that action before. I will evaluate when I am next at my computer

**Comment by b3vis** [comment]
> The eval error looks legit.

I agree, I’ll see if I can add a fix to this PR before the CI goes live if this gets merged

**Comment by Mic92** [comment]
I can merge this when the ci runs through. I assume the current error we are seeing is due to the nixpkgs bump?

**Comment by b3vis** [comment]
> I can merge this when the ci runs through. I assume the current error we are seeing is due to the nixpkgs bump?

Looks that way, I will double check and resolve so we can get this merged

**Comment by b3vis** [comment]
@Mic92, Trick was to upgrade the flakes nixpkgs input to 21.11, now it runs `nix flake check` without error. I've also included a updated flake lock file

**Comment by nzbr** [comment]
The CI pipeline has been replaced in significant parts since the PR was created, so the changes introduced here aren't really applicable anymore

**Pull request closed without merging**